opm/io/eclipse/rst/well.cpp

```diff
@@ -167,66 +183,81 @@ RstWell::RstWell(const ::Opm::UnitSystem& unit_system,
-                 const std::vector<double>& rseg) :
-    RstWell(unit_system, header, group_arg, zwel, iwel, swel, xwel, icon, scon, xcon)
+Opm::RestartIO::RstWell::RstWell(const UnitSystem&          unit_system,
+                                 const RstHeader&           header,
+                                 const std::string&         group_arg,
+                                 const std::string*         zwel,
+                                 const int*                 iwel,
+                                 const float*               swel,
+                                 const double*              xwel,
+                                 const int*                 icon,
+                                 const float*               scon,
+                                 const double*              xcon,
+                                 const std::vector<int>&    iseg,
+                                 const std::vector<double>& rseg)
+    : RstWell { unit_system, header, group_arg,
+                zwel, iwel, swel, xwel,
+                icon, scon, xcon }
 {
+    if (this->msw_index == 0) {
+        return;
+    }
 
-    if (this->msw_index) {
-        std::unordered_map<int, std::size_t> segment_map;
-        for (int is=0; is < header.nsegmx; is++) {
-            std::size_t iseg_offset = header.nisegz * (is + (this->msw_index - 1) * header.nsegmx);
-            std::size_t rseg_offset = header.nrsegz * (is + (this->msw_index - 1) * header.nsegmx);
-            auto other_segment_number = iseg[iseg_offset + VI::ISeg::SegNo];
-            if (other_segment_number != 0) {
-                auto segment_number = is + 1;
-                segment_map.insert({segment_number, this->segments.size()});
-                this->segments.emplace_back( unit_system, segment_number, iseg.data() + iseg_offset, rseg.data() + rseg_offset);
-            }
+    std::unordered_map<int, std::size_t> segment_map;
+    for (int is = 0; is < header.nsegmx; ++is) {
+        const std::size_t iseg_offset = header.nisegz * (is + (this->msw_index - 1)*header.nsegmx);
+        const std::size_t rseg_offset = header.nrsegz * (is + (this->msw_index - 1)*header.nsegmx);
+        const auto other_segment_number = iseg[iseg_offset + VI::ISeg::SegNo];
+
+        if (other_segment_number != 0) {
+            auto segment_number = is + 1;
+            segment_map.insert({segment_number, this->segments.size()});
+            this->segments.emplace_back(unit_system, segment_number,
+                                        iseg.data() + iseg_offset,
+                                        rseg.data() + rseg_offset);
         }
+    }
 
-        for (auto& segment : this->segments) {
-            if (segment.outlet_segment != 0) {
-                auto& outlet_segment = this->segments[ segment_map[segment.outlet_segment] ];
-                outlet_segment.inflow_segments.push_back(segment.segment);
-            }
+    for (auto& segment : this->segments) {
```

Comment:
can be const.

Reply:
> can be const.

Yep, you're right&ndash;good catch.  I've pushed an update to that effect.